_re2.cc

```diff
@@ -314,12 +320,81 @@ static PyTypeObject RegexpSet_Type2 = {
   regexp_set_new,                  /*tp_new*/
 };
 
+// getters for MatchObject2
+static PyObject*
+match_pos_get(MatchObject2* self)
+{
+  return PyLong_FromSsize_t(self->pos);
+}
+
+static PyObject*
+match_endpos_get(MatchObject2* self)
+{
+  return PyLong_FromSsize_t(self->endpos);
+}
+
+static PyObject*
+match_re_get(MatchObject2* self)
+{
+  Py_INCREF(self->re);
+  return self->re;
+}
+
+static PyObject*
+match_string_get(MatchObject2* self)
+{
+  Py_INCREF(self->string);
+  return self->string;
+}
+
+// getters for RegexObject2
+static PyObject*
+regexp_groups_get(RegexpObject2* self)
+{
+  return PyLong_FromSsize_t(self->groups);
+}
+
+static PyObject*
+regexp_groupindex_get(RegexpObject2* self)
+{
+  if (self->groupindex == NULL) {
+	PyObject* groupindex = PyDict_New();
+	if (groupindex == NULL) {
+	  return NULL;
+	}
+
+	const std::map<std::string, int>& name_map = self->re2_obj->NamedCapturingGroups();
+	for (std::map<std::string, int>::const_iterator it = name_map.begin(); it != name_map.end(); ++it) {
+	  // This used to return an int() on Py2, but now returns a long() to be
+	  // consistent across Py3 and Py2.
+	  PyObject* index = PyLong_FromLong(it->second);
+	  if (index == NULL) {
+		break;
+	  }
+	  int res = PyDict_SetItemString(groupindex, it->first.c_str(), index);
+	  Py_DECREF(index);
+	  if (res < 0) {
+		break;
+	  }
+    }
+    Py_INCREF(groupindex);
+    self->groupindex = groupindex;
+  }
+
+  return self->groupindex;
+}
+
+static PyObject* regexp_pattern_get(RegexpObject2* self)
+{
+  return self->pattern;
```

Comment:
Why doesn't this incref?

Reply:
Hm, you are correct this should increase the reference count. Not sure why/how I missed it but I'll fix it!